crates/cubecl-cpp/src/cuda/dialect.rs

```diff
@@ -89,8 +98,37 @@ impl DialectTypes<Self> for CudaDialect {
         scalars: &[(Elem<Self>, usize)],
         flags: &Flags,
     ) -> std::fmt::Result {
+        // All FP4/FP6/FP8 elems map to the same type, so we need to deduplicate them
+        let mut items_deduplicated = HashSet::new();
+
+        for item in items {
+            let mut item = *item;
+            match item.elem() {
+                Elem::FP4(_) => {
+                    item.elem = Elem::FP4(FP4Kind::E2M1);
+                }
+                Elem::FP4x2(_) => {
+                    item.elem = Elem::FP4x2(FP4Kind::E2M1);
```

Comment:
I'm unsure if we want to have `fp4x2` or any `x2` types. For the same reason, we don't have `f16x2` and `bf16x2`. We have those in the C++ compiler, of course, but not on the frontend side since vectorization is programmed with the line size instead. I guess the reason you included it was to read and write values with 8 bits, but maybe we could use something else instead? The concept of a minimum line size, to which we have to pad if it's uneven, would make sense for reading data back from the GPU.

Reply:
This comment is on the C++ dialect, which is where we do have `F16x2` 😅 I'm assuming you're talking about the type though. I only added that to allow `bytemuck` to properly convert buffers without losing semantics like would if we just have it represented as u8. It doesn't implement `CubePrimitive` so it's not used on the GPU side, only on CPU for conversion.

Reply:
Yeah I was refering to the type declaration in `cubecl_common`. Do you think we could make a type like `Packed<e2m1, 2>` to be used with all subtypes ?

```rust
pub struct Packed<E: SubByteElement, const FACTOR: usize> {
    data: [E; FACTOR],
}
```
where `core::mem::size_of(data) >= 1`

Reply:
The problem is that we can't really declare something like that in current Rust. You can't declare sub-byte types at all, so `fp4x2` would be 2 bytes with a type like that. I'll look into this more when I work on the sub-byte migration PR.

Reply:
I might look into https://crates.io/crates/packed_struct to see if we can do something with that.